lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -827,6 +833,21 @@ OpFoldResult AndOp::fold(FoldAdaptor adaptor) {
     }
   }
 
+  // x0 = icmp(x, y)
+  // x1 = n_icmp(x, y)
+  // and(x0, x1) -> 0
+  for (auto [index, lhs] : llvm::enumerate(getOperands())) {
+    for (auto rhs : getOperands().drop_front(index)) {
+      auto lhsOp = lhs.getDefiningOp<ICmpOp>();
+      auto rhsOp = rhs.getDefiningOp<ICmpOp>();
+      if (lhsOp && rhsOp && canCombineOppositeCmpIntoConstant(lhsOp, rhsOp)) {
```

Comment:
Could you hoist the condition for `lhsOp`? 
```suggestion
    if (auto lhsOp = lhs.getDefiningOp<ICmpOp>(); lhsOp && lhs.getTwoState() ){
    for (auto rhs : getOperands().drop_front(index)) {
      auto rhsOp = rhs.getDefiningOp<ICmpOp>();
      if (rhsOp && canCombineOppositeCmpIntoConstant(lhsOp, rhsOp)) {
```

Reply:
Resolved as the new implementation is per-op not per two ops.